Source/UnitTests/Common/SettingsHandlerTest.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <gtest/gtest.h>
+
+#include "Common/SettingsHandler.h"
+
+namespace
+{
+constexpr size_t BUFFER_SIZE = Common::SettingsHandler::SETTINGS_SIZE;
+
+// The encrypted bytes corresponding to the following settings, in order:
+//   "key" = "val"
+std::array<u8, 9> BUFFER_CONTENTS_A = {(u8)'\x91', (u8)'\x91', (u8)'\x90', (u8)'\xEE', (u8)'\xD1',
+                                       (u8)'/',    (u8)'\xF0', (u8)'4',    (u8)'y'};
+
+// The encrypted bytes corresponding to the following settings, in order:
+//   "key1" = "val1"
+//   "key2" = "val2"
+//   "foo" = "bar"
+std::array<u8, 31> BUFFER_CONTENTS_B = {
+    (u8)'\x91', (u8)'\x91', (u8)'\x90', (u8)'\xE2', (u8)'\x9A', (u8)'8', (u8)'\xFD', (u8)'U',
+    (u8)'B',    (u8)'\xEA', (u8)'\xC4', (u8)'\xF6', (u8)'^',    (u8)'\xF', (u8)'\xDF', (u8)'\xE7',
+    (u8)'\xC3', (u8)'\n',   (u8)'\xBB', (u8)'\x9C', (u8)'P',    (u8)'\xB1', (u8)'\x10', (u8)'\x82',
+    (u8)'\xB4', (u8)'\x8A', (u8)'\r',   (u8)'\xBE', (u8)'\xCD', (u8)'r',    (u8)'\xF4'};
+}
+
+TEST(SettingsHandlerTest, EncryptSingleSetting)
+{
+  Common::SettingsHandler handler;
+  handler.AddSetting("key", "val");
+  Common::SettingsHandler::Buffer buffer = handler.GetBytes();
+
+  ASSERT_EQ(buffer.size(), BUFFER_SIZE);
+  for (size_t i = 0; i < BUFFER_CONTENTS_A.size(); ++i)
+  {
+    EXPECT_EQ(buffer[i], BUFFER_CONTENTS_A[i]);
+  }
+}
+
+TEST(SettingsHandlerTest, DecryptSingleSetting)
+{
+  Common::SettingsHandler::Buffer buffer;
+  std::copy(BUFFER_CONTENTS_A.begin(), BUFFER_CONTENTS_A.end(), buffer.begin());
+  Common::SettingsHandler handler(std::move(buffer));
```

Comment:
It's unfortunate that this little "dance" is needed. I would suggest adding an additional constructor to `SettingsHandler` so it accepts `const Buffer&` in addition to `Buffer&&`.

Reply:
Two thoughts here:

1) The little "dance" is necessary not just because of the move constructor, but because a `Buffer` always has length 256. If we take a const ref, we'd still have to `std::copy` from our `BUFFER_CONTENTS` array into the passed-in `Buffer` somewhere, or add a bunch of padding to our hard-coded arrays.

2) Would the additional constructor be useful in the prod code? If not, and if we always want to avoid a copy when constructing a `SettingsHandler` in the prod code, my inclination would be to avoid adding it just for the sake of making the tests a bit more concise. I'm inspired here by the "DAMP, not DRY" philosophy in https://abseil.io/resources/swe-book/html/ch12.html, and a culture in my day job of avoiding adding test-only API surface when possible. (I don't want to apply style guidance I learned at my day job if the Dolphin project has a different philosophy, though.)

WDYT? I'll defer to your judgment!

Reply:
Actually, now that I think about more, I don't think moving a `std::array` of primitives actually avoids any copies (https://stackoverflow.com/a/28988913). So maybe this class should only have a constructor that takes `const Buffer&`, if a `Buffer` will always just be a type alias for an array.

I'll do whatever you think is appropriate in this PR, and I'm happy to put together a follow-up PR if we want to remove the rvalue refs in favor of const refs.

Reply:
Sigh, I read the `std::array` spec more closely and realized my point 1 above isn't true either. We can have the hardcoded arrays be length 256 and use list initialization to explicitly initialize their first few elements and value-initialize the rest. And then we can simply use the assignment operator instead of `std::copy` in the test bodies.

In the latest commit, the "dance" to create a temp buffer has been reduced to `Buffer buffer = BUFFER_A`. Seems sufficiently clean that there's no need to add a new constructor, but I'm happy to do it if you want.

Sorry about all the mistakes, and thanks for bearing with me!

Reply:
Sorry, I missed these comments. Let's go with what you have in the current version.